### PR TITLE
Expect UnauthorizedAccessException in tvOS if extracting fifo from tar without elevation

### PR DESCRIPTION
**Pull request by carlossanlop**
I saw a tvOS failure in [one of my PRs](https://dev.azure.com/dnceng/public/_build/results?buildId=1840021&view=ms.vss-test-web.build-test-results-tab&runId=48593844&resultId=100888&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab). I had not seen it before, so I'm unsure if it's intermittent or consistent.

The `ExtractToFile_SpecialFile_Unelevated_Throws` method threw `UnauthorizedAccessException` when attempting to extract the fifo from the archive into disk:

https://github.com/dotnet/runtime/blame/051b4828c7d3a0cad3289830ef9fd2120f45bb2b/src/libraries/System.Formats.Tar/tests/TarReader/TarReader.ExtractToFile.Tests.Unix.cs#L39

In other Unix platforms, extracting fifos does not throw, but apparently it does on tvOS. 

Here's the callstack:

```
[14:55:07.7170830] 2022-06-22 14:55:07.706 System.Formats.Tar.Tests[58680:78698627]    Exception messages: System.UnauthorizedAccessException : Access to the path '/private/var/mobile/Containers/Data/Application/11374FD2-B51C-49C4-844D-F40F062ACE94/tmp/1pslf5q1.gf2/output' is denied.
[14:55:07.7171130] ---- System.IO.IOException : Operation not permitted
[14:55:07.7189880] 2022-06-22 14:55:07.708 System.Formats.Tar.Tests[58680:78698627]    Exception stack traces:    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
[14:55:07.7190220]    at System.Formats.Tar.TarEntry.ExtractAsFifo(String destinationFileName)
[14:55:07.7190300] 2022-06-22 14:55:07.708 System.Formats.Tar.Tests[58680:78698627]    at System.Formats.Tar.TarEntry.ExtractToFileInternal(String filePath, String linkTargetPath, Boolean overwrite)
[14:55:07.7190360]    at System.Formats.Tar.TarEntry.ExtractToFile(String destinationFileName, Boolean overwrite)
[14:55:07.7201670] 2022-06-22 14:55:07.709 System.Formats.Tar.Tests[58680:78698627]    at System.Formats.Tar.Tests.TarReader_ExtractToFile_Tests.ExtractToFile_SpecialFile_Unelevated_Throws()
```

### CODE REVIEW

**Comment by danmoseley** [comment]
Instead of wondering whether it's consistent or not, I suggest to leave the code as it is but wrap in a try catch for UnauthorizedAccessException and in the catch simply Assert that you're on tvOS.

File: src/libraries/System.Formats.Tar/tests/TarReader/TarReader.TarEntry.ExtractToFile.Tests.Unix.cs [hunk 1]
```diff
@@ -10,36 +10,35 @@ namespace System.Formats.Tar.Tests
     public partial class TarReader_TarEntry_ExtractToFile_Tests : TarTestsBase
     {
         [PlatformSpecific(TestPlatforms.AnyUnix & ~TestPlatforms.tvOS)] // https://github.com/dotnet/runtime/issues/68360
-        [Fact]
-        public async Task SpecialFile_Unelevated_Throws_Async()
+        [ConditionalFact(nameof(IsUnixButNotSuperUser))]
+        public void SpecialFile_Unelevated_Throws()
```
Comment (carlossanlop) [inline src/libraries/System.Formats.Tar/tests/TarReader/TarReader.TarEntry.ExtractToFile.Tests.Unix.cs:14]: I noticed this test, which was supposed to be the synchronous one, was written as the asynchronous one.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @carlossanlop

File: src/libraries/System.Formats.Tar/tests/TarReader/TarReader.TarEntry.ExtractToFileAsync.Tests.Unix.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ namespace System.Formats.Tar.Tests
     public partial class TarReader_TarEntry_ExtractToFileAsync_Tests : TarTestsBase
     {
         [PlatformSpecific(TestPlatforms.AnyUnix & ~TestPlatforms.tvOS)] // https://github.com/dotnet/runtime/issues/68360
-        [Fact]
+        [ConditionalFact(nameof(IsUnixButNotSuperUser))]
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply src/libraries/System.Formats.Tar/tests/TarReader/TarReader.TarEntry.ExtractToFileAsync.Tests.Unix.cs:13]: The Unix can be removed, yes.

**Comment by carlossanlop** [comment]
The test failures are unrelated, which means the change worked. I can remove the `Unix` portion of the condition in my next PR with test fixes, @adamsitnik .

File: src/libraries/System.Formats.Tar/tests/TarReader/TarReader.TarEntry.ExtractToFileAsync.Tests.Unix.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamsitnik) [reply src/libraries/System.Formats.Tar/tests/TarReader/TarReader.TarEntry.ExtractToFileAsync.Tests.Unix.cs:13]: @carlossanlop you are right, I am sorry I've missed the negation

**Pull request merged**